src/main/java/org/jenkinsci/plugins/workflow/steps/EchoStep.java

```diff
@@ -36,19 +40,43 @@
  */
 public class EchoStep extends Step {
 
-    private final String message;
+    private static final int MAX_LABEL_LENGTH = 100;
+
+    private String message;
+    private String label;
 
     @DataBoundConstructor
     public EchoStep(String message) {
         this.message = message;
     }
 
+    @DataBoundSetter
+    public void setMessage(String message) {
+        this.message = message;
+    }
+
     public String getMessage() {
         return message;
     }
 
+    @DataBoundSetter
+    public void setLabel(String label) {
+        if (label == null)
+            throw new IllegalArgumentException();
```

Comment:
It feels a little awkward to me to throw an exception here. On the one hand it makes sense to fail fast, but I don't think anyone should be calling this method directly except for Stapler and maybe devs in a unit test, so perhaps adding an `@Nonnull` annotation would be good enough, and you still have to have the null check in `start` anyway for builds deserialized from before the label existed so it seems like a bit of a wash. Maybe you could add a `doCheckLabel` method to the descriptor to do some validation for anyone using the pipeline syntax page, but I'm not sure whether that works well with optional `@DataBoundSetter`s.

Reply:
@dwnusbaum
Yeah, since I check for null in the `start` this is overkill.  I like the `@NonNull` annotation for self-documenting code.